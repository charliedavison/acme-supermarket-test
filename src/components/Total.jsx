import React from 'react';
import PropTypes from 'prop-types';

class Total extends React.Component {
    constructor(props) {
        super(props);
    };

    render() {
        return (
            <div className="total">
                <div>{`Subtotal: £${this.props.subtotal}`}</div>
                <div>{`Discounts: -£${parseFloat((this.props.subtotal - this.props.total).toFixed(2))}`}</div>
                <div className="text-bold">{`Total: £${this.props.total}`}</div>
            </div>
        );
    }
};
Total.defaultProps = {
    subtotal: 0.0,
    total: 0.0
};
Total.propTypes = {
    subtotal: PropTypes.number.isRequired,
    total: PropTypes.number.isRequired
};
module.exports = Total;