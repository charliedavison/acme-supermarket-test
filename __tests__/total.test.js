import React from 'react';
import { shallow } from 'enzyme';
import Total from './../src/components/Total.jsx';

describe('<Total />', () => {

    test('Should match snapshot', () => {
        const component = shallow(<Total />);
        expect(component).toMatchSnapshot();
    });

    test('Should show correct totals where discount has been applied', () => {
        const props = {
            subtotal: 20,
            total: 13
        };
        const component = shallow(<Total {...props} />);

        expect(component.find('.total').childAt(0).text()).toContain(`£20`);
        expect(component.find('.total').childAt(1).text()).toContain(`-£7`);
        expect(component.find('.total').childAt(2).text()).toContain(`£13`);
    });

    test('Should show correct totals where no discount has been applied', () => {
        const props = {
            subtotal: 20,
            total: 20
        };
        const component = shallow(<Total {...props} />);

        expect(component.find('.total').childAt(0).text()).toContain(`£20`);
        expect(component.find('.total').childAt(1).text()).toContain(`-£0`);
        expect(component.find('.total').childAt(2).text()).toContain(`£20`);
    });
});