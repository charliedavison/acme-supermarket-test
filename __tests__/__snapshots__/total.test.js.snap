// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Total /> Should match snapshot 1`] = `
<div
  className="total"
>
  <div>
    Subtotal: £0
  </div>
  <div>
    Discounts: -£0
  </div>
  <div
    className="text-bold"
  >
    Total: £0
  </div>
</div>
`;
